src/ethernet_interface.cpp

```diff
@@ -468,20 +514,44 @@ bool EthernetInterface::dhcp6(bool value)
     return value;
 }
 
+void EthernetInterface::deleteStaticIPv4Addresses()
+{
+    std::unique_ptr<IPAddress> ptr;
+    for (auto it = addrs.begin(); it != addrs.end();)
+    {
+        if ((it->second->origin() == IP::AddressOrigin::Static) &&
+            (it->second->type() == IP::Protocol::IPv4))
+        {
+            ptr = std::move(it->second);
+            it = addrs.erase(it);
+            writeConfigurationFile();
+            manager.get().reloadConfigs();
+        }
+        else
+        {
+            it++;
+        }
+    }
+}
+
 EthernetInterface::DHCPConf EthernetInterface::dhcpEnabled(DHCPConf value)
 {
-    auto old4 = EthernetInterfaceIntf::dhcp4();
+
+    // TODO This is a workaround to avoid IPv4 static and DHCP IP address
+    // coexistence
+    if ((value == DHCPConf::v4) || (value == DHCPConf::both))
+    {
+        // Delete all IPv4 static addresses while enabling DHCP
+        deleteStaticIPv4Addresses();
+    }
+
     auto new4 = EthernetInterfaceIntf::dhcp4(value == DHCPConf::v4 ||
                                              value == DHCPConf::both);
-    auto old6 = EthernetInterfaceIntf::dhcp6();
     auto new6 = EthernetInterfaceIntf::dhcp6(value == DHCPConf::v6 ||
```

Comment:
What are these new4 and new6 are used for here? This is common code for IPv6 as well. So i think better to cover IPv6 side of DHCP things as well with this patch